sdk/search/Azure.AI.Search/src/SearchExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+namespace Azure.AI.Search
+{
+    /// <summary>
+    /// Search extension methods.
+    /// </summary>
+    internal static partial class SearchExtensions
+    {
+        /// <summary>
+        /// Assert that the given URI uses HTTPS as its scheme.
+        /// </summary>
+        /// <param name="endpoint">The URI to validate.</param>
+        /// <param name="paramName">
+        /// The name of the parameter for this URI, to use with an
+        /// <see cref="ArgumentException"/>.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the <paramref name="endpoint"/> is not using HTTPS as
+        /// its scheme.
+        /// </exception>
+        public static void AssertHttpsScheme(this Uri endpoint, string paramName)
+        {
+            Debug.Assert(endpoint != null);
+            if (!string.Equals(endpoint.Scheme, Uri.UriSchemeHttps, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException($"{paramName} only supports https.", paramName);
```

Comment:
Nit: might want to capitalize "HTTPS".

Reply:
Changing to `Uri.UriSchemeHttps` in the error message.